components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -9,6 +9,16 @@ use crate::codec::data_type::*;
 use crate::codec::mysql::json::*;
 use crate::Result;
 
+#[inline]
+fn encode_to_json_path(path: &Option<Bytes>) -> Result<Option<PathExpression>> {
```

Comment:
Is parse_json_path_expr used separately elsewhere? If not, why not integrate the match statement into parse_json_path_expr directly?